### ISSUE TITLE
Recurse option (-R) does not include files in current directory

### ISSUE THREAD

**Issue by tinyapps** [Parchive/par2cmdline#150]
As shown in the following example, par2cmdline skips files in the current directory (in this case, foo.txt) when run with the `-R` option (Recurse into subdirectories); i.e., it _only_ processes files in subdirectories, ignoring files in the current directory:

```
~/test$ tree
.
├── foo.txt
└── sub1
    ├── bar.txt
    └── sub2
        └── baz.txt
```

```
~/test$ par2 c -R *

Block size: 4
Source file count: 2
Source block count: 6
Recovery block count: 1
Recovery file count: 1

Opening: sub1/sub2/baz.txt
Opening: sub1/bar.txt
Computing Reed Solomon matrix.
Constructing: done.
Wrote 4 bytes to diskets
Writing recovery packets
Writing verification packets
Done
```

Is this by design? Shouldn't files in the current directory also be included?

Also, if there is nothing in the current directory other than subdirectories, `par2 c -R *` returns "_You must specify a list of files when creating._" even though any files in the current directory are ignored.

**Comment by boggeroff** [comment]
Is the PAR3 thingy so exciting that bugs like these are of no interest to the maintainers?

**Comment by mdnahas** [comment]
@boggeroff I agree that this does sound like a bug and it is probably important to some users.  We are all volunteers here.  We work on what we want to.  I encourage you to join us and fix it yourself.  It looks like a good one to start on.  

Or, if you cannot code and it is important to you, you can always hire someone to fix it.  

Or, if you have no money, you can _try_ to inspire our volunteers to work on it, but I suggest asking nicely in a message laden with pathos.   "Starving children depend on this bug being fixed" "I lost all my data and I want to prevent the other many users of this feature from suffering the same tragic fate", "I'm trying to convince my company to use Par2 and we need this feature to do this awesome thing", etc..   Emotion motivates people more than a call to "do their duty".  

Or, you can try one of the other clients that write Par2 files.

BTW, I don't think any of the people involved in the Par3 discussion would call themselves "maintainers" of par2cmdline.  Two of them maintain their own clients.  I designed the Par2 file format; I did not write par2cmdline.  I'm working on the Par3 file format because I overlooked a bug in the Par2 file format and am trying to fix my mistake.  

I would love to have you join us and fix this bug.  We also have other bugs you can work on.   If you say "but I don't want to work on other bugs", you will find yourself in the same situation as all the volunteers who work on this project.  Please treat them respectfully.  If you've found par2cmdline useful, I also suggest thanking them.

**Comment by boggeroff** [comment]
@mdnahas Thank you for you response! I know I might have sounded a bit too negative, I intented to get insight into the heartbeat of the project. In my opinion, fault tolerance in data archiving with the least amount of investment is a really important topic, and I don't see proportional aweareness of this by the OS community. Therefore, I'm a little frustrated to begin with (why is this overlooked?). The par2 format seems to be the only solution. And the par2cmdline seems to be the only reliable project to help in practice.

Now that you mention financial support, what are the options? Having my external consciousness protected from rotting worth spending a little money... Would hardly be the first occasion I contributed to OSS either. Actually, right now, I feel motivated to start a C++17 rewrite of this project, with Python CLI and distrbution. We'll see if that sticks.

Anyways, I have deep respect for the par* endeavours and generally for those who contribute things like this to the public.
And I'm going to try that blueberry pie recipe (no joke).

**Comment by BlackIkeEagle** [comment]
> As shown in the following example, par2cmdline skips files in the current directory (in this case, foo.txt) when run with the `-R` option (Recurse into subdirectories); i.e., it _only_ processes files in subdirectories, ignoring files in the current directory:
> 
> ```
> ~/test$ tree
> .
> ├── foo.txt
> └── sub1
>     ├── bar.txt
>     └── sub2
>         └── baz.txt
> ```
> 
> ```
> ~/test$ par2 c -R *
> 
> Block size: 4
> Source file count: 2
> Source block count: 6
> Recovery block count: 1
> Recovery file count: 1
> 
> Opening: sub1/sub2/baz.txt
> Opening: sub1/bar.txt
> Computing Reed Solomon matrix.
> Constructing: done.
> Wrote 4 bytes to diskets
> Writing recovery packets
> Writing verification packets
> Done
> ```
> 
> Is this by design? Shouldn't files in the current directory also be included?

No it is not by design, but the behaviour is actually correct.

due to your shell expanding the `*` to end up with the following command `par2 c -R foo.txt sub1` by design `par2` will use foo.txt as the par filename, in your case resulting in only the subfolders being covered by the recovery files.

Changing this would cause a lot of tools using par2 to potentially have issues. So Not actually going to 'fix' this issue.

> 
> Also, if there is nothing in the current directory other than subdirectories, `par2 c -R *` returns "_You must specify a list of files when creating._" even though any files in the current directory are ignored.

Do you actually mean subdirectories with files? or empty subdirectories

**Comment by BlackIkeEagle** [comment]
Should be covered by documentation now
